diesel/src/sqlite/types/json.rs

```diff
@@ -0,0 +1,575 @@
+    fn test_write_jsonb_false() {
+        let value = serde_json::Value::Bool(false);
+        let mut buffer = Vec::new();
+        write_jsonb_value(&value, &mut buffer).unwrap();
+        assert_eq!(buffer, vec![JSONB_FALSE]);
+    }
+
+    #[test]
+    fn test_write_jsonb_int() {
+        let value = serde_json::Value::Number(serde_json::Number::from(1));
+        let mut buffer = Vec::new();
+        write_jsonb_value(&value, &mut buffer).unwrap();
+        assert_eq!(buffer, vec![create_header(JSONB_INT, 0x01), b'1']);
+    }
+
+    #[test]
+    fn test_write_jsonb_float() {
+        let value = serde_json::Value::Number(serde_json::Number::from_f64(1.5).unwrap());
+        let mut buffer = Vec::new();
+        write_jsonb_value(&value, &mut buffer).unwrap();
+        assert_eq!(
+            buffer,
+            vec![create_header(JSONB_FLOAT, 0x03), b'1', b'.', b'5']
+        );
+    }
+
+    #[test]
+    fn test_write_jsonb_text() {
+        let value = serde_json::Value::String("foo".to_string());
+        let mut buffer = Vec::new();
+        write_jsonb_value(&value, &mut buffer).unwrap();
+        assert_eq!(
+            buffer,
+            vec![create_header(JSONB_TEXT, 0x03), b'f', b'o', b'o']
+        );
+    }
+
+    #[test]
+    fn test_write_jsonb_array() {
+        let value = json!([1, true]);
+        let mut buffer = Vec::new();
+        write_jsonb_value(&value, &mut buffer).unwrap();
+        assert_eq!(
+            buffer,
+            vec![
+                create_header(JSONB_ARRAY, 0x03), // Array header
+                create_header(JSONB_INT, 0x01),   // Integer header
+                b'1',                             // Integer payload "1"
+                create_header(JSONB_TRUE, 0x00),  // Boolean header for "true"
+            ]
+        );
+    }
+
+    #[test]
+    fn test_write_jsonb_object() {
+        let value = json!({"key": 42});
+        let mut buffer = Vec::new();
+        write_jsonb_value(&value, &mut buffer).unwrap();
+        assert_eq!(
+            buffer,
+            vec![
+                create_header(JSONB_OBJECT, 0x07), // Object header
+                create_header(JSONB_TEXT, 0x03),   // Key header: "key"
+                b'k',
+                b'e',
+                b'y',                           // Key: "key"
+                create_header(JSONB_INT, 0x02), // Value header: integer "42"
+                b'4',
+                b'2', // Value: 42
+            ]
+        );
+    }
+
+    #[test]
+    fn json_to_sql() {
+        let conn = &mut connection();
+        let res = diesel::select(json!(true).into_sql::<Json>().eq(&sql("json('true')")))
+            .get_result::<bool>(conn);
+        assert!(res.unwrap());
+    }
+
+    #[test]
+    fn jsonb_to_sql() {
+        let conn = &mut connection();
+        let res = diesel::select(json!(true).into_sql::<Jsonb>().eq(&sql("jsonb('true')")))
+            .get_result::<bool>(conn);
+        assert!(res.unwrap());
+    }
```

Comment:
This test is a good starting point, but I would like to see some more variants of this for:

* All variants of `serde_json::Value`
* Some edgecases including escaped string values and new lines

Reply:
> Some edgecases including escaped string values and new lines

@weiznich done.